actions/merkleroots/get_merkleroots_endpoint_test.go

```diff
@@ -0,0 +1,171 @@
+package merkleroots_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet/actions/testabilities"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/stretchr/testify/require"
+)
+
+const merklerootsURL = "/api/v1/merkleroots"
+
+func setupWalletAndClientForUser(t *testing.T, responseCode *int, response *string) (*testabilities.SPVWalletApplicationAssertions, func(), *resty.Client) {
+	given, then := testabilities.New(t)
+	cleanup := given.StartedSPVWallet()
+	client := given.HttpClient().ForUser()
+
+	if response != nil && responseCode != nil {
+		given.BHS().WillRespondForMerkleRoots(*responseCode, *response)
+	}
+
+	return &then, cleanup, client
+}
+
+func TestGETMerkleRootsSuccess(t *testing.T) {
+
+	testCases := map[string]struct {
+		query            string
+		expectedResponse models.MerkleRootsBHSResponse
+	}{
+		"Get MerkleRoots success no query params": {
+			query: "",
+			expectedResponse: models.MerkleRootsBHSResponse{
+				Content: testabilities.MockedBHSData,
+				Page: models.ExclusiveStartKeyPageInfo{
+					TotalElements:    len(testabilities.MockedBHSData),
+					Size:             len(testabilities.MockedBHSData),
+					LastEvaluatedKey: "",
+				},
+			},
+		},
+		"Get MerkleRoots success with last evaluated key param": {
+			query: "?lastEvaluatedKey=df2b060fa2e5e9c8ed5eaf6a45c13753ec8c63282b2688322eba40cd98ea067a",
+			expectedResponse: models.MerkleRootsBHSResponse{
+				Content: testabilities.MockedBHSData[5:],
+				Page: models.ExclusiveStartKeyPageInfo{
+					TotalElements:    len(testabilities.MockedBHSData),
+					Size:             len(testabilities.MockedBHSData[5:]),
+					LastEvaluatedKey: "",
+				},
+			},
+		},
+	}
+
+	for name, tt := range testCases {
+		t.Run(name, func(t *testing.T) {
+
+			// given
+			expResponseJSON, err := json.Marshal(tt.expectedResponse)
+			require.NoError(t, err, "Failed to marshall expected response")
+			then, cleanup, client := setupWalletAndClientForUser(t, nil, nil)
+			defer cleanup()
+			url := merklerootsURL
+
+			if tt.query != "" {
+				url = url + tt.query
+			}
+
+			// when
+			res, err := client.R().
+				SetHeader("Content-Type", "application/json").
+				Get(url)
+			require.NoError(t, err, "Unexpected error occurred while getting MerkleRoots")
```

Comment:
it is not needed for those tests, please read the documentation of given.HttpClient method, if it's not enaugh clear, then read below.

The given.HttpClient method is adding error handler to resty so that any unexpected error that could be returned here will fail the tests so you don't need to make any assumptions about returned error and can focus only on res.